File: src/main/kotlin/october/october07/P3273.java
package october.october07;

import java.util.*;
import java.io.*;

public class P3273 {
    static int N,X, ans;
    static int arr[];

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        N = Integer.parseInt(br.readLine());
        arr = new int[N];
        HashSet<Integer> set = new HashSet<>();
        StringTokenizer st = new StringTokenizer(br.readLine());
        for(int i=0; i<N; i++) {
            arr[i] = Integer.parseInt(st.nextToken());
            set.add(arr[i]);
        }
        X = Integer.parseInt(br.readLine());

        for(int i=0; i<N; i++) {
            if(set.contains(X-arr[i])) {
                ans++;
            }
        }
        System.out.println(ans/2);
    }
}

File: src/main/kotlin/october/october07/P3273.kt
package october.october07

private fun main() {
    val n = readLine()!!.toInt()
    val a = readLine()!!.split(" ").map{ it.toInt() }
    val x = readLine()!!.toInt()
    val f = a.groupBy{ it }.map{ (a, b) -> a to b.size }.toMap()
    println(f.map{ (y, c) ->  c.toLong() * f.getOrElse(x - y) { 0 } }.sum() / 2)
}